bin/eslint.js

```diff
@@ -143,8 +143,9 @@ ${getErrorMessage(error)}`;
         console.warn("You can also run this command directly using 'npm init @eslint/config'.");
 
         const spawn = require("cross-spawn");
+        const npm = process.env.npm_execpath || "npm";
 
-        spawn.sync("npm", ["init", "@eslint/config"], { encoding: "utf8", stdio: "inherit" });
+        spawn.sync(npm, ["init", "@eslint/config"], { encoding: "utf8", stdio: "inherit" });
```

Comment:
I want to note that `bun`'s `init` command is not the same as npm. similar for deno (i don't know if they set npm_execpath or not, and what it is set to)

so this solution wont achieve the desired fix.

Reply:
Looks like this is much more complicated than just using `npm_execpath`, as we'd need to figure out what exactly we're running and update the command accordingly.

Since the `--init` option is just a convenience, perhaps we could leave this as-is and just update the info message (`You can also run this command directly using 'npm init @eslint/config'.`) to list commands for other package managers?

Reply:
@paperdave  I tested locally, directly modifying eslint through the `node_modules` folder did work!

Reply:
> Since the `--init` option is just a convenience, perhaps we could leave this as-is and just update the info message (`You can also run this command directly using 'npm init @eslint/config'.`) to list commands for other package managers?

I think that's probably the best path forward here. If we want, at some point, to investigate what it would take to have ESLint do this automatically, then we probably want to do so through an RFC to really understand what it is we're trying to accomplish and how we can sufficiently abstract runtime information throughout the project.

Reply:
I also think it would be better to leave things like they are for now. ESLint only supports Node.js officially, and Node.js users should always have `npm` installed. Conversely, some Bun users don't have `node` or `npm` installed on their machines. We can consider adding support for Bun, but that would definitely need an RFC.